### PR TITLE
mgmt, bug fix on disk encryption and vault URI, improve on VM update

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
track1 commit https://github.com/Azure/azure-libraries-for-java/pull/1355/files

track1 commit https://github.com/Azure/azure-libraries-for-java/pull/1354/commits/a91bf8a83c6cebb815455f890968e08ab42c5ee2

track1 commit https://github.com/Azure/azure-libraries-for-java/pull/1350

1. bug fix on `getByResourceGroupAsync(rgName, vmName, InstanceViewTypes.INSTANCE_VIEW)`
2. bug fix on `VirtualMachineEncryptionConfiguration.keyvaultUrl` when it is not global cloud
3. force deletion for resource group
4. avoid the PATCH request in VM update flow, if not required

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/ProxyEncryptionMonitorImpl.java [hunk 1]
```diff
@@ -83,19 +83,22 @@ public DiskVolumeEncryptionMonitor refresh() {
                 .flatMap(
                     virtualMachine -> {
                         VirtualMachineExtensionInner extension = encryptionExtension(virtualMachine);
-                        if (EncryptionExtensionIdentifier.isNoAADVersion(osType(), extension.typeHandlerVersion())) {
-                            self.resolvedEncryptionMonitor =
-                                (osType() == OperatingSystemTypes.LINUX)
+                        if (extension != null) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/ProxyEncryptionMonitorImpl.java:86]: NPE

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/LinuxDiskVolumeNoAADEncryptionMonitorImpl.java [hunk 2]
```diff
@@ -125,7 +126,8 @@ public DiskVolumeEncryptionMonitor refresh() {
      * @return the retrieved virtual machine
      */
     private Mono<VirtualMachineInner> retrieveVirtualMachineAsync() {
-        return computeManager.serviceClient().getVirtualMachines().getByResourceGroupAsync(rgName, vmName);
+        return computeManager.serviceClient().getVirtualMachines()
+            .getByResourceGroupAsync(rgName, vmName, InstanceViewTypes.INSTANCE_VIEW);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/LinuxDiskVolumeNoAADEncryptionMonitorImpl.java:130]: existing bug in track2 (track1 is correct)

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/LinuxVMDiskEncryptionConfiguration.java [hunk 3]
```diff
@@ -3,18 +3,54 @@
...
+    /**
+     * Creates LinuxVMDiskEncryptionSettings.
+     *
+     * @param keyVaultId the resource ID of the KeyVault to store the disk encryption key
+     * @param vaultUri URI of the key vault data-plane endpoint
+     * @param aadClientId client ID of an AAD application which has permission to the KeyVault
+     * @param aadSecret client secret corresponding to the client ID
+     */
+    public LinuxVMDiskEncryptionConfiguration(String keyVaultId,
+                                              String vaultUri,
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/LinuxVMDiskEncryptionConfiguration.java:33]: Recommend customer to provide the `vaultUri`.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineEncryptionConfiguration.java [hunk 4]
```diff
@@ -90,8 +105,16 @@ public String keyVaultId() {
...
+            return vaultUri;
+        }
+
+        String keyVaultDnsSuffix =
+            (azureEnvironment == null || CoreUtils.isNullOrEmpty(azureEnvironment.getKeyVaultDnsSuffix()))
+                ? AzureEnvironment.AZURE.getKeyVaultDnsSuffix()
+                : azureEnvironment.getKeyVaultDnsSuffix();
         String keyVaultName = ResourceUtils.nameFromResourceId(this.keyVaultId);
-        return String.format("https://%s.vault.azure.net/", keyVaultName.toLowerCase(Locale.ROOT));
+        return String.format("https://%1$s%2$s", keyVaultName.toLowerCase(Locale.ROOT), keyVaultDnsSuffix);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/VirtualMachineEncryptionConfiguration.java:117]: fix here for non global cloud.

**Review by weidongxu-microsoft — COMMENTED** [review]
Also regenerate resources for force deletion.

**Review by ChenTanyi — APPROVED** [review]

**Pull request merged**